rclpy/rclpy/node.py

```diff
@@ -368,8 +381,27 @@ def create_publisher(
             failed = True
         if failed:
             self._validate_topic_or_service_name(topic)
-        publisher = Publisher(publisher_handle, msg_type, topic, qos_profile, self.handle)
+
+        publisher = Publisher(
+            publisher_handle, msg_type,
+            topic, callback_group, qos_profile, event_callbacks, self.handle)
         self.publishers.append(publisher)
+        callback_group.add_entity(publisher)
+
+        if event_callbacks:
+            if event_callbacks.deadline_callback:
```

Comment:
These need to be kept track of better so that they can be removed when pub is removed